File: src/main/java/equidiaryDB/services/LoginService.java
package equidiaryDB.services;

import equidiaryDB.EquidiaryDB;
import io.javalin.http.Context;
import org.json.JSONObject;

import java.sql.SQLException;

public class LoginService {

    private static final String USERNAME = "username";
    private static final String PASSWORD = "password";

    public static void login(Context context) throws SQLException {
        JSONObject requestBody = new JSONObject(context.body());
        String username = requestBody.getString(USERNAME);
        String password = requestBody.getString(PASSWORD);

        int userId = EquidiaryDB.db.isCorrectUser(username, password);

        if (isCorrectUser(userId)) {
            context.result("Hello " + username);
        } else {
            context.result("Wrong username or password");
        }
    }

    private static boolean isCorrectUser(int userId) {
        return userId >= 0;
    }
}

File: src/main/java/equidiaryDB/services/LoginService.kt
package equidiaryDB.services

import equidiaryDB.EquidiaryDB.db
import io.javalin.http.Context
import io.javalin.http.Handler
import org.json.JSONObject

class LoginService : Handler {
    private val USERNAME = "username"
    private val PASSWORD = "password"

    override fun handle(context: Context) {
        val requestBody = JSONObject(context.body())
        val username = requestBody.getString(USERNAME)
        val password = requestBody.getString(PASSWORD)
        val userId = db.isCorrectUser(username, password)
        if (isCorrectUser(userId)) {
            context.result("Hello $username")
        }
        else {
            context.result("Wrong username or password")
        }
    }

    private fun isCorrectUser(userId: Int): Boolean {
        return userId >= 0
    }
}
